### PR TITLE
feat(date-picker): add action buttons and slot

### LINKED ISSUE #1273: [datePicker&timePicker] add action buttons
e.g
image

### PR DESCRIPTION
**Pull request by Sixlty**
## Checklist

List of tasks you have already done and plan to do.

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

Describe your modifications here.
1.add default buttons(confirm and cancel)
2.add slot named action

## Issues

The issues you want to close, formatted as close #1273.

## Related Links

Links related to this pr.

### CODE REVIEW

**Comment by chouchouji** [comment]
可以完善下类型文件，
image

**Comment by Sixlty** [comment]
> 可以完善下类型文件， image

File: packages/varlet-ui/src/date-picker/DatePicker.vue [hunk 1]
```diff
@@ -96,6 +96,17 @@
...
     </div>
+    <div :class="n('date-action')" v-if="showAction">
+      <slot name="action">
+        <var-button text type="default" @click="onCancel">
+          {{ pack.pickerCancelButtonText }}
+        </var-button>
+
+        <var-button text type="primary" @click="onConfirm">
+          {{ pack.pickerConfirmButtonText }}
+        </var-button>
```
Comment (haoziqaq) [inline packages/varlet-ui/src/date-picker/DatePicker.vue:107]: The component's internationalized text should be independent and should not use picker definitions

File: packages/varlet-ui/src/date-picker/DatePicker.vue [hunk 2]
```diff
@@ -96,6 +96,17 @@
         />
       </transition>
     </div>
+    <div :class="n('date-action')" v-if="showAction">
```
Comment (haoziqaq) [inline packages/varlet-ui/src/date-picker/DatePicker.vue:99]: rename actions, ref 
image

File: packages/varlet-ui/src/date-picker/props.ts [hunk 3]
```diff
@@ -87,7 +87,13 @@ export const props = {
     type: Boolean,
     default: true,
   },
+  showAction: {
```
Comment (haoziqaq) [inline packages/varlet-ui/src/date-picker/props.ts:90]: showActions

File: packages/varlet-ui/src/date-picker/DatePicker.vue [hunk 4]
```diff
@@ -120,6 +131,7 @@ import { isArray, toNumber, doubleRaf, call } from '@varlet/shared'
 import { createNamespace, formatElevation } from '../utils/components'
 import { padStart } from '../utils/shared'
 import { pack } from '../locale'
+import VarButton from '../button'
```
Comment (chouchouji) [inline packages/varlet-ui/src/date-picker/DatePicker.vue:134]: Remember to import less file of button

File: packages/varlet-ui/src/date-picker/date-picker.less [hunk 5]
```diff
@@ -48,7 +49,7 @@
   font-size: var(--date-picker-font-size);
   position: relative;
   min-width: var(--date-picker-min-width);
-  height: var(--date-picker-height);
+  // height: var(--date-picker-height);
```
Comment (BeADre) [inline packages/varlet-ui/src/date-picker/date-picker.less:52]: There will be an inappropriate height.

image

File: packages/varlet-ui/src/date-picker/DatePicker.vue (hunk 1 above)
Comment (Sixlty) [reply packages/varlet-ui/src/date-picker/DatePicker.vue:107]: ok

File: packages/varlet-ui/src/date-picker/DatePicker.vue (hunk 4 above)
Comment (Sixlty) [reply packages/varlet-ui/src/date-picker/DatePicker.vue:134]: 好的

File: packages/varlet-ui/src/date-picker/date-picker.less (hunk 5 above)
Comment (Sixlty) [reply packages/varlet-ui/src/date-picker/date-picker.less:52]: If using a fixed height may not be conducive to the expansion of custom action panels, then I'll try fixing the body height of the date-picker

File: packages/varlet-ui/src/date-picker/DatePicker.vue (hunk 4 above)
Comment (Sixlty) [reply packages/varlet-ui/src/date-picker/DatePicker.vue:134]: 原本已有导入less文件
image

Comment (chouchouji) [reply packages/varlet-ui/src/date-picker/DatePicker.vue:134]: 好的

File: packages/varlet-ui/src/date-picker/date-picker.less (hunk 5 above)
Comment (BeADre) [reply packages/varlet-ui/src/date-picker/date-picker.less:53]: Any way can be, but the height can't change, you can see [this](https://play.vuetifyjs.com/#eNptj0EOwiAQRa9Cxq2liXajwSaew7pAOibYOhCgNcZ4dym1SRtdAX8eH97pBd6p/Ggt7zuEPYiAd9vKgGVFjIk+U4aC1IQuBSly5sFunQ/6+jxU4K1UmElnOqor+FKJq2NNZrVq0DFssZdBG4o3NsWMi+T0JFtJNSB+NkxFl0ClkqSwFfl4+gOY5nco8oXOGC0+NlnFNGqN0nE/t16UwHsNW77lOxjWghdw/gD2tGMR)

https://github.com/varletjs/varlet/assets/34639100/2285131d-9b66-4930-958b-a246444f9a31

Comment (Sixlty) [reply packages/varlet-ui/src/date-picker/date-picker.less:53]: I get it

File: packages/varlet-ui/src/date-picker/date-picker.less [hunk 6]
```diff
@@ -2,7 +2,7 @@
   --date-picker-border-radius: 4px;
   --date-picker-font-size: var(--font-size-md);
   --date-picker-min-width: 290px;
-  --date-picker-height: 385px;
+  // --date-picker-height: 385px;
```
Comment (BeADre) [inline packages/varlet-ui/src/date-picker/date-picker.less:5]: remove comments and update docs(new props, slot , cssvar and removed cssvar)

**Comment by Sixlty** [comment]
@BeADre ok的，本来也只是为了提供一些便利而已

**Comment by BeADre** [comment]
@haoziqaq  你review一下 要是没问题就可以合了

**Comment by haoziqaq** [comment]
好



---Original---
From: ***@***.***&gt;
Date: Wed, Dec 6, 2023 10:01 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [varletjs/varlet] feat(date-picker): add action buttons and slot(PR #1321)




 
@haoziqaq  你review一下 要是没问题就可以合了

File: packages/varlet-ui/src/date-picker/__tests__/index.spec.js [hunk 7]
```diff
@@ -424,3 +424,18 @@ test('test datePicker titleColor', async () => {
...
+    },
+    props: {
+      showActions: true,
+    },
+  })
+
+  expect(wrapper.find('.var-date-picker__actions').text()).toBe('This is action slot')
+
+  wrapper.unmount()
+})
```
Comment (haoziqaq) [inline packages/varlet-ui/src/date-picker/__tests__/index.spec.js:441]: 这里是不是不需要了

**Comment by haoziqaq** [comment]
@Sixlty  感谢您的贡献，抱歉占用了您不少的时间。

**Comment by Sixlty** [comment]
感谢各位指导意见，非常荣幸能为优秀项目贡献绵薄之力

**Pull request merged**